kaldb/src/main/java/com/slack/kaldb/clusterManager/ReplicaCreatorService.java

```diff
@@ -39,23 +48,46 @@ protected void startUp() throws Exception {
     metadataStoreService.awaitRunning(DEFAULT_START_STOP_DURATION);
 
     replicaMetadataStore = new ReplicaMetadataStore(metadataStoreService.getMetadataStore(), true);
-    replicaMetadataStore.addListener(replicaNodeListener());
+
+    snapshotMetadataStore =
+        new SnapshotMetadataStore(metadataStoreService.getMetadataStore(), true);
+    snapshotMetadataStore.addListener(this::createReplicasForUnassignedSnapshots);
+
+    // we need to fire this at startup, so any snapshots that were created while we were offline are
+    // handled
+    createReplicasForUnassignedSnapshots();
   }
 
   @Override
   protected void shutDown() throws Exception {
     LOG.info("Closing replica create service");
 
     replicaMetadataStore.close();
+    snapshotMetadataStore.close();
```

Comment:
Since we're moving these stores up a level and passing them in, it won't be necessary to do anything special here. We also don't need to un-register listeners explicitly, as this is already handled as as part of the abstract base class. 

https://github.com/slackhq/kaldb/blob/master/kaldb/src/main/java/com/slack/kaldb/metadata/core/CacheableMetadataStore.java#L81